spec/cyanobyte.schema.json

```diff
@@ -0,0 +1,216 @@
+        "address": {
+          "description": "I2C Address of the component",
+          "type": "integer"
+        },
+        "addressMask": {
+          "description": "Address mask for addresses that can be altered"
+        }
+      },
+      "required": [
+        "addressType",
+        "address",
+        "addressMask"
+      ]
+    },
+    "registers": {
+      "description": "Chip registers",
+      "type": "array",
+      "items": {
+        "description": "Outer object",
+        "type": "object",
+        "patternProperties": {
+          "^.*$": {
+            "description": "Chip register",
+            "type": "object",
+            "properties": {
+              "title": {
+                "description": "Title of the register",
+                "type": "string"
+              },
+              "description": {
+                "description": "Description of the register",
+                "type": "string"
+              },
+              "address": {
+                "description": "Address of the register",
+                "type": "integer"
+              },
+              "length": {
+                "description": "Length of the register in bits.",
+                "type": "integer"
+              }
+            },
+            "required": [
+              "title",
+              "description",
+              "address",
+              "length"
+            ]
+          }
+        },
+        "additionalProperties": false
+      }
+    },
+    "functions": {
+      "description": "Chip functions",
+      "type": "array",
+      "items": {
+        "description": "Outer object",
+        "type": "object",
+        "patternProperties": {
+          "^.*$": {
+            "description": "Chip function",
+            "type": "object",
+            "properties": {
+              "title": {
+                "description": "Title of the function",
+                "type": "string"
+              },
+              "description": {
+                "description": "Description of the function",
+                "type": "string"
+              },
+              "register": {
+                "description": "Register of the function",
+                "type": "string"
+              },
+              "readWrite": {
+                "description": "Can you read or write to this register",
+                "type": "string",
+                "enum": [
+                  "R",
+                  "R/W",
+                  "W",
+                  "n"
+                ]
+              },
+              "bitStart": {
+                "description": "What is the starting bit?",
+                "type": "integer"
+              },
+              "bitEnd": {
+                "description": "What is the ending bit?",
+                "type": "integer"
+              },
+              "type": {
+                "description": "What type of value is this?",
+                "type": "string",
+                "enum": [
+                  "enum"
```

Comment:
Are we going to add `twos-complement`, like in the `tempBoundary` case?

Reply:
We need to create a list of what we want to support. Was just trying to cover the bare minimum here until we had the conversation.

Reply:
Will file an issue to track